src/tree/updater_quantile_hist.cc

```diff
@@ -740,19 +761,34 @@ void QuantileHistMaker::Builder<GradientSumT>::InitSampling(const std::vector<Gr
       const size_t ibegin = tid * discard_size;
       const size_t iend = (tid == (nthread - 1)) ?
                           info.num_row_ : ibegin + discard_size;
-
-      rnds[tid].discard(discard_size * tid);
+      constexpr uint64_t kBase = 16807;
```

Comment:
Is this arbitrary?

Reply:
No. it's similar value as for minstd_rand: https://en.cppreference.com/w/cpp/numeric/random/linear_congruential_engine
By requirements for lcg we have to use primitive root modulo n.

Reply:
yes. I applied refactoring for subsampling, thank for proposal!